hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/DNRPCLoadGenerator.java

```diff
@@ -0,0 +1,155 @@
+import picocli.CommandLine.Option;
+
+import java.util.List;
+import java.util.concurrent.Callable;
+
+import static org.apache.hadoop.ozone.common.PayloadUtils.generatePayloadBytes;
+
+/**
+ * Utility to generate RPC request to DN.
+ */
+@Command(name = "dn-echo",
+        aliases = "dne",
+        description =
+                "Generate echo RPC request to DataNode",
+        versionProvider = HddsVersionProvider.class,
+        mixinStandardHelpOptions = true,
+        showDefaultValues = true)
+public class DNRPCLoadGenerator extends BaseFreonGenerator
+        implements Callable<Void> {
+
+  private static final int RPC_PAYLOAD_MULTIPLICATION_FACTOR = 1024;
+  private static final int MAX_SIZE_KB = 2097151;
+  private Timer timer;
+  private OzoneConfiguration configuration;
+  private ByteString payloadReqBytes;
+  private int payloadRespSize;
+  private XceiverClientSpi client;
+  @Option(names = {"--payload-req"},
+          description =
+                  "Specifies the size of payload in KB in RPC request. " +
+                          "Max size is 2097151 KB",
+          defaultValue = "0")
+  private int payloadReqSizeKB = 0;
+
+  @Option(names = {"--payload-resp"},
+          description =
+                  "Specifies the size of payload in KB in RPC response. " +
+                          "Max size is 2097151 KB",
+          defaultValue = "0")
+  private int payloadRespSizeKB = 0;
+
+  @Option(names = {"--container-id"},
+      description = "Send echo to DataNodes associated with this container")
+  private long containerID;
+
+  @Option(names = {"--sleep-time-ms"},
+      description = "Let DataNode to pause for a duration (in milliseconds) for each request",
+      defaultValue = "0")
+  private int sleepTimeMs = 0;
+
+  @CommandLine.ParentCommand
+  private Freon freon;
+
+  // empy constructor for picocli
+  DNRPCLoadGenerator() {
+  }
+
+  @VisibleForTesting
+  DNRPCLoadGenerator(OzoneConfiguration ozoneConfiguration) {
+    this.configuration = ozoneConfiguration;
+  }
+
+  @Override
+  public Void call() throws Exception {
+    Preconditions.checkArgument(payloadReqSizeKB >= 0,
+            "OM echo request payload size should be positive value or zero.");
+    Preconditions.checkArgument(payloadRespSizeKB >= 0,
+            "OM echo response payload size should be positive value or zero.");
+
+    if (configuration == null) {
+      configuration = freon.createOzoneConfiguration();
+    }
+    ScmClient scmClient = new ContainerOperationClient(configuration);
+    ContainerInfo containerInfo = scmClient.getContainer(containerID);
+
+    List<Pipeline> pipelineList = scmClient.listPipelines();
+    Pipeline pipeline = pipelineList.stream()
+        .filter(p -> p.getId().equals(containerInfo.getPipelineID()))
+        .findFirst()
+        .orElse(null);
+    XceiverClientFactory xceiverClientManager =
+        new XceiverClientManager(configuration);
+    client = xceiverClientManager.acquireClient(pipeline);
```

Comment:
11k is an interesting number, we should support an option `--clients`, this would enable multiple clients via multiple threads, to max out the Datanode and or the client as well.